modules/kubernetes-addons/vpa/locals.tf

```diff
@@ -1,71 +1,38 @@
- * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
- * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
- */
-
 locals {
+  name                 = "vpa"
   service_account_name = "vpa-sa"
-  namespace            = "vpa"
 
   default_helm_config = {
-    name                       = "vpa"
-    chart                      = "vpa"
-    repository                 = "https://charts.fairwinds.com/stable"
-    version                    = "1.0.0"
-    namespace                  = "vpa"
-    timeout                    = "1200"
-    create_namespace           = true
-    description                = "Kubernetes Vertical Pod Autoscaler"
-    lint                       = false
-    wait                       = true
-    wait_for_jobs              = false
-    verify                     = false
-    keyring                    = ""
-    repository_key_file        = ""
-    repository_cert_file       = ""
-    repository_ca_file         = ""
-    repository_username        = ""
-    repository_password        = ""
-    disable_webhooks           = false
-    reuse_values               = false
-    reset_values               = false
-    force_update               = false
-    recreate_pods              = false
-    cleanup_on_fail            = false
-    max_history                = 0
-    atomic                     = false
-    skip_crds                  = false
-    render_subchart_notes      = true
-    disable_openapi_validation = false
-    dependency_update          = false
-    replace                    = false
-    postrender                 = ""
-    set                        = []
-    set_sensitive              = []
-    values                     = local.default_helm_values
+    name        = local.name
+    chart       = local.name
+    repository  = "https://charts.fairwinds.com/stable"
+    version     = "1.0.0"
+    namespace   = local.name
+    description = "Kubernetes Vertical Pod Autoscaler"
+    values      = local.default_helm_values
+    timeout     = "1200"
   }
 
+  default_helm_values = [templatefile("${path.module}/values.yaml", {
+    sa-name = local.service_account_name
+  })]
+
   helm_config = merge(
     local.default_helm_config,
     var.helm_config
   )
 
-  default_helm_values = [templatefile("${path.module}/values.yaml", {
-    vpa_sa_name = local.service_account_name
-  })]
+  irsa_config = {
+    kubernetes_namespace              = local.helm_config["namespace"]
+    kubernetes_service_account        = local.service_account_name
+    create_kubernetes_namespace       = true
+    create_kubernetes_service_account = true
+    iam_role_path                     = "/"
+    tags                              = var.tags
+    eks_cluster_id                    = var.eks_cluster_id
+    irsa_iam_policies                 = var.irsa_policies
+    irsa_iam_permissions_boundary     = var.irsa_permissions_boundary
+  }
```

Comment:
I don't think we need IRSA for VPA. Are you creating any policies here?

Reply:
addressed.